File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/AssistantCreateRequest.kt
package com.sdkwork.ai

data class AssistantCreateRequest(
    val model: String? = null,
    val name: String? = null,
    val description: String? = null,
    val instructions: String? = null,
    val tools: List<Tool>? = null,
    val metadata: Map<String, String>? = null,
    val temperature: Double? = null,
    val toolResources: Map<String, Any>? = null,
    val responseFormat: ResponseFormat? = null,
    val topP: Double? = null
)

File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/models/assistant_create_request.py
from dataclasses import dataclass
from typing import Optional, List, Dict, Any

@dataclass
class AssistantCreateRequest:
    """Assistant creation request"""
    model: str
    name: str = None
    description: str = None
    instructions: str = None
    tools: List[Tool] = None
    metadata: Dict[str, str] = None
    temperature: float = None
    tool_resources: Dict[str, Any] = None
    response_format: ResponseFormat = None
    top_p: float = None
